homeassistant/components/xiaomi_aqara.py

```diff
@@ -189,23 +195,33 @@ def remove_device_service(call):
 class XiaomiDevice(Entity):
     """Representation a base Xiaomi device."""
 
-    def __init__(self, device, name, xiaomi_hub):
+    def __init__(self, device, name, hass, xiaomi_hub):
         """Initialize the xiaomi device."""
         self._state = None
+        self._lock = threading.Lock()
+        self._hass = hass
+        self._is_available = True
         self._sid = device['sid']
         self._name = '{}_{}'.format(name, self._sid)
         self._write_to_hub = xiaomi_hub.write_to_hub
         self._get_from_hub = xiaomi_hub.get_from_hub
         self._device_state_attributes = {}
+        self._remove_unavailability_tracker = None
+        self._track_unavailable()
```

Comment:
Don't call this from the constructor but instead use the hook `async_added_to_hass` (it's a coroutine). `self.hass` will be available at that point.

Reply:
Ok, it's the same issue as https://github.com/home-assistant/home-assistant/pull/11631#discussion_r162252995.